src/test/run-pass/futures-api.rs

```diff
@@ -32,8 +35,8 @@ impl Future for MyFuture {
     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         // Wake twice
         let waker = cx.waker();
-        waker.wake();
-        waker.wake();
+        waker.wake_by_ref();
+        waker.wake_by_ref();
```

Comment:
```suggestion
        waker.wake();
```
From a superficial glance at why `wake_by_ref` was added, wasn't the point to avoid unnecessary `Arc` count traffic?

Reply:
Yes-- that's exactly what's achieved here by using `wake_by_ref`. Changing this to `wake()` would first require a `cx.waker().clone()`, which is what we want to avoid.

Reply:
Ah nvm, I didn't realize `waker` returns a reference.